index.js

```diff
@@ -109,23 +109,24 @@ Auth0Metrics.prototype.setUserId = function(uid) {
 
 
 Auth0Metrics.prototype.identify = function (id, traits, callback) {
-  var segment = this.segment();
+  var args = [].slice.call(arguments);
 
-  if(typeof id !== 'string'){
-    traits = id;
-    id = null;
-  }
+  // Argument reshuffling.
+  if (_.isFunction(traits)) callback = traits, traits = null;
+  if (_.isObject(id)) traits = id, id = null;
+
+  var segment = this.segment();
 
   if (segment.loaded) {
-    if(null != id){
-      this.setUserId(id);
+    if (null != id) this.setUserId(id);
 
-      try {
-        segment.identify(id, traits);
-      } catch (error) {
-        debug('segment analytics error: %o', error);
-      }
+    try {
+      if (_.isFunction(args[args.length - 1])) args.pop();
+      segment.identify.apply(segment, args);
+    } catch (error) {
+      debug('segment analytics error: %o', error);
     }
+
   }else{
     debug('identify call without segment');
```

Comment:
We should have an Identify endpoint in the DWH for only traits, right @sicarul? For A/B testing it'd be good to not loose users. Could we add this?